### PR TITLE
Light sensor button string

### PR DESCRIPTION
**Pull request by aninternetof**
This adds blocks for:
- Reading left and right light sensor values
- Reading whether button A or button B has been pressed
- Writing a message to the micro:bit 5x5 display

I used the block generator tool to create each of these.
image

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,5 +1,14 @@
 # Blockly [![Build Status]( https://travis-ci.org/google/blockly.svg?branch=master)](https://travis-ci.org/google/blockly)
 
+## Quick Usage
+
+```bash
+> n lts
+> npm install
+> npm run build
+> google-chrome tests/rovercode_playground.html
+> # import tests/rovercode-all.xml. Generate javascript and compare to tests/rovercode-all-output.js
```
Comment (aninternetof) [inline README.md:10]: I figured out how to use their unit testing system. It's really cool.
image
But unfortunately, there's no support for mocking. Our blocks are not very functional, so there wasn't really any way to assert on what any of them returned.

As some assurance that I've tried out these blocks, you can load up the `rovercode-all.xml` program in the playground and assure that its output matches `rovercode-all-output.js`. I can work on some way for the CI to do this.

File: blocks/sensors.js [hunk 2]
```diff
@@ -48,3 +48,28 @@ Blockly.Blocks['sensors_get_covered'] = {
...
+    this.setHelpUrl("https://docs.rovercode.com/blocks/light-sensor-value");
+  }
+};
+
+Blockly.Blocks['button_press'] = {
+  init: function() {
+    this.appendDummyInput()
+        .appendField("button")
+        .appendField(new Blockly.FieldDropdown([["A","A"], ["B","B"]]), "BUTTON")
+        .appendField("has been pressed");
```
Comment (aninternetof) [inline blocks/sensors.js:69]: We have to make a decision how to handle the button state. The micro:bit only provides the `pressed` action, so we can't keep track of pressed/not-pressed in the UI.
I propose that we set the state to `hasBeenPressed` when a button is pressed, and we keep it like that until this block queries its state, when we'll clear it.

File: generators/javascript/display.js [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+ */
+'use strict';
+
+goog.provide('Blockly.JavaScript.display');
+
+goog.require('Blockly.JavaScript');
+
+Blockly.JavaScript['display_message'] = function(block) {
+  var value_message = Blockly.JavaScript.valueToCode(block, 'MESSAGE', Blockly.JavaScript.ORDER_ATOMIC);
+  var code = 'displayMessage(String('+ value_message +'));\n';
```
Comment (aninternetof) [inline generators/javascript/display.js:34]: This seems to work fine on `String("foo")`, `String(1234)`, and `String(true)`.

File: README.md (hunk 1 above)
Comment (aninternetof) [reply README.md:10]: image

```javascript
var a;


setMotor('LEFT', 'FORWARD', '99');
setMotor('LEFT', 'FORWARD', '');
stopMotor('LEFT');
a = (getLightSensorValue('LEFT'));
a = (buttonHasBeenPressed('A'));
displayMessage(String('Sup'));
displayMessage(String(123));
displayMessage(String(true));

```

File: blocks/sensors.js [hunk 4]
```diff
@@ -48,3 +48,28 @@ Blockly.Blocks['sensors_get_covered'] = {
     this.setHelpUrl('http://www.example.com/');
   }
 };
+
+Blockly.Blocks['light_sensor_value'] = {
+  init: function() {
+    this.appendDummyInput()
+        .appendField(new Blockly.FieldDropdown([["left","LEFT"], ["right","RIGHT"]]), "SENSORS")
```
Comment (cabarnes) [inline blocks/sensors.js:55]: Should this be singular `SENSOR`?

File: generators/javascript/display.js (hunk 3 above)
Comment (cabarnes) [reply generators/javascript/display.js:34]: What about a variable?

File: blocks/sensors.js (hunk 4 above)
Comment (aninternetof) [reply blocks/sensors.js]: Good catch. Fixed.

File: generators/javascript/display.js (hunk 3 above)
Comment (aninternetof) [reply generators/javascript/display.js:34]: Added that to the rovercode-all.xml. Seems to work well.

Comment (aninternetof) [reply generators/javascript/display.js:34]: `displayMessage(String(a));`

**Pull request merged**